src/main/java/org/opensearch/ad/MemoryTracker.java

```diff
@@ -238,42 +235,66 @@ public long estimateTotalModelSize(AnomalyDetector detector, int numberOfTrees,
      *      - SmallNodeStore (small node store since our sample size is 256, less than the max of short): 6120
      *      + BoxCacheFloat
      *        - other: 104
-     *        - BoundingBoxFloat: (1040 + 255* (dimension * 4 * 2 + 64)) * adjusted bounding box cache usage,
-     *           where if full we have 255 inner node and each box has 80 bytes.
-     *           Plus metadata, we can have in total 21544 bytes.
-     *           {@code adjusted bounding box cache usage = (bounding box cache fraction >= 0.3? 1: bounding box cache fraction)}
-     *           {@code >= 0.3} we will still initialize bounding box cache array of the max size,
-     *           but exclude them using the cache ratio.  It is not guaranteed we will only
-     *           use cache ratio in the array.  For example, with cache ratio 0.5, we used 150
-     *           out of 255 elements.  So can have two float array whose size is the number of
-     *           dimensions; other constants are the metadata size.
+     *        - BoundingBoxFloat: (1040 + 255* (dimension * 4 + 16) * 2 + 32)) * actual bounding box cache usage,
+     *           {@code actual bounding box cache usage = (bounding box cache fraction >= 0.3? 1: bounding box cache fraction)}
+     *           {@code >= 0.3} we will still initialize bounding box cache array of the max size.
      * In total, RCF size is
-     *  56 + # trees * (2248 + 152 + 6120 + 104 + (1040 + 255* (dimension * 4 * 2 + 64)) * adjusted bounding box cache ratio) +
-     *  (256 * # trees  * 2 + 256 * # trees * dimension) * 4 bytes  * 0.5 + 1064 + 24 + 24 + 16
-     *  = 56 + # trees * (8624 + (1040 + 255 * (dimension * 8 + 64)) * adjusted bounding box cache ratio) + 256 * # trees *
-     *   (3 + dimension) * 4 * 0.5 + 1128
+     *  56 + # trees * (2248 + 152 + 6120 + 104 + (1040 + 255* (dimension * 4 + 16) * 2 + 32)) * adjusted bounding box cache ratio) +
```

Comment:
This is a lot to follow. Just wondering, do you have any tables of experimental versus predicted (with this equation) data size for (T)RCFModelSize

Reply:
yes, https://gist.github.com/kaituo/77b985921eda631a534157e8e11f6c57

Reply:
Nice, those look pretty good. How accurate does this estimate need to be? In other words, whats the margin for error?

Reply:
We don't have a formal margin.  My mental margin is 10%.